VueJS SPA

Akirathan opened the pull request:
Closes https://github.com/enso-org/enso/issues/10212

Replaces the statically generated web pages by a VueJS single-page-application.

Docs in [README.md](https://github.com/enso-org/engine-benchmark-results/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5).

Currently testing the deployment in the forked repo https://github.com/Akirathan/engine-benchmark-results

Commit:
Remove the statically generated HTML pages.

Commit:
Add .gitignore

Commit:
Init vite vue

Commit:
Add construct_index.py helper script

Commit:
Create cache/index.json

Commit:
Add .vscode dir

Commit:
Remove styles.css from vite template

Commit:
example startDate and endDate is current from 05/2024

Commit:
Update README

Commit:
Merge branch 'main' into vue

Commit:
Regenerate cache/index.json

Commit:
Update README

Commit:
Add cache schema and validation

Commit:
Ensure cache is valid before generating the website

Commit:
Add schema version

Commit:
Add schema directory

Commit:
Add schema version 2.0

Commit:
Add schema CHANGELOG

Commit:
Configure fs-server as vite plugin

Commit:
vuetify is a vite-plugin

Commit:
Display a simple bench graph

Commit:
Install linter and prettier

Commit:
format

Commit:
Implement DatapointStore.findDataPointsByLabel

Commit:
BenchFilter is initialized with selected startDate and endDate

Commit:
DEFAULT_DAYS_TO_FETCH is an env constant

Commit:
Display data for all labels

Commit:
File fetching is asynchronous

Commit:
Do not store duplicate data

Commit:
Implement updateBenchData

Commit:
Ensure datapoints are not duplicated in the store

Commit:
BenchGraph has min and max date on x axis

Commit:
Display only MAX_LABELS on MainPage

Commit:
Fix commit message selection

Commit:
fmt

Commit:
maxDate is fixed to 2024-06-01

Commit:
Add space between graph and selection details

Commit:
Add benchRunURL to selection

Commit:
Add filter by labels (classes)

Commit:
Link in bench run URL is displayed only if some elemnt is selected

Commit:
Selected element is emphasized

Commit:
MainPage contains only BenchSearch

Commit:
BenchGraphs loads all graphs

Commit:
Remove unused variables from MainPage

Commit:
fmt

Commit:
Add BenchmarksInfo

Commit:
Display list of benchmarks that violate six-sigma

Commit:
datapointStore stores points by label

Commit:
updateBenchData is not async

Commit:
startDate and endDate is provided as route

Commit:
Fix type error in BenchGraph

Commit:
Fix display of BenchGraph is too small

Commit:
Add some docs to README

Commit:
Spaces between BenchGraph

Commit:
Remove construct_index.py script.

This is not necessary - it is already integrated in latest develop.

Commit:
Revert "startDate and endDate is provided as route"

This reverts commit 27d883658d69305e4c3b24b3d54a318c5dd4bbce.

Commit:
Remove local index.json

Commit:
Merge branch 'main' into vue

# Conflicts:
#	README.md
#	engine-benchs.html
#	index.html
#	stdlib-benchs.html

Commit:
fmt

Commit:
Label filtering by Engine benchmarks

Commit:
Refactor some stores to use HashMaps for better performance

Commit:
Process single file is async

Commit:
Introduce cacheFileStore

Commit:
Add functionality to display all engine benchmarks and refactor label handling

Commit:
Label selection is handled by BenchSearch component

Commit:
Use light red color for the clear selection button

Commit:
Pinia stores are used only in setup section of a component

Commit:
Remove assert from addDatapoint.

This was causing huge performance overhead.

Commit:
Revert "Pinia stores are used only in setup section of a component"

This reverts commit 3b4a45a73f66abdb0f3f0f438483a57f7c52a35a.

Commit:
fmt

Commit:
BenchmarkGraph is wider

Commit:
fmt

Commit:
datapoints are sorted by datetime

Commit:
BenchGraphs reacts on changes on dates and labels

Commit:
Use console.debug instead of console.log

Commit:
Unify handling of new props changed on BenchGraphs

Commit:
Clear selection when props changed

Commit:
Add docs

JaroslavTulach commented on stdlib-benchs.html:
I don't think this file should be deleted. It should be kept and remain updated daily as it was by now:
- we don't know if the new infrastructure is good enough replacement of the old
- we don't want to loose existing URL references to the old pages

Akirathan replied:
Reverted in 4e9f6bf8928a721a590347bb14f07b19694d84e7

JaroslavTulach commented on index.html:
```diff
@@ -1,51 +1,12 @@
-  <div class="container">
-    <h1>Engine and Standard libs benchmarks</h1>
-    <ul>
-      <li><a href="engine-benchs.html">Engine Benchmarks</a></li>
```
These old links should be kept until we decide the new infrastructure is a complete and good enough replacement for the old one. You can move the `<div class="container">` to the bottom of the `index.html` - it doesn't have to take a prominent place.

Akirathan replied:
The static websites are reintroduced in 4e9f6bf8928a721a590347bb14f07b19694d84e7 from the previous [merge with main](c61396ec0c314b16b13e96dbbad46fa3a3ca4635).

JaroslavTulach commented on schema/cache-v2.json:
```diff
@@ -0,0 +1,177 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$comment": "Version 1.0",
```
The file is named `v2` and yet the version is `1.0` - isn't that inconsistent?

Akirathan replied:
Good point. Bumped the version in f1cd95fd6dc0bef0b85a339273d5495b94bd1db2

JaroslavTulach commented on scripts/validate_cache.js:
```diff
@@ -0,0 +1,26 @@
+const ajv = new Ajv()
+addFormats(ajv)
+
+const schemaPath = './schema/cache-v1.json'
```
Why do we have `cache-v2.json` when we validate against version 1?

JaroslavTulach replied:
Who executes this file?

Akirathan replied:
So far, nobody. The `cache-v2.json` schema corresponds to the new schema introduced in https://github.com/enso-org/enso/pull/10224. This is just a preparation for when https://github.com/enso-org/enso/pull/10224 gets integrated (I want to integrate only the new data schema, not the rest of the changes in that PR).

Commit:
Add gh-pages workflow

JaroslavTulach commented on src/utils/data.ts:
```diff
@@ -0,0 +1,184 @@
+import { useDataPointStore } from '../stores/dataPointsStore'
+import { useLabelStore } from '../stores/labelStore'
+
+export interface Commit {
```
These `interface`s seem to describe the structure of JSON files that are being loaded in. How are they connected to the schema definitions? Why do we have TypeScript definition and also JSON schema definition?

Having two independent definition is good only for one thing: to make them inconsistent.

Akirathan replied:
Let's deal with this schema-related issues once we tackle the new data in https://github.com/enso-org/enso/pull/10224

Commit:
Update vite-config.

- Copy cache dir inside dist
- Update URLs to point to the served repo
- MIN_DATE is a global constant

JaroslavTulach commented on src/vite-env.d.ts:
```diff
@@ -0,0 +1,11 @@
+/// <reference types="vite/client" />
+
+declare const FS_URL: string
```
Why is this called `FS`?

Where will the `FS_URL` point to on the website? I am not able to find that out...

Akirathan replied:
The name may be a bit misleading. In dev mode, `FS_URL` points to a simple server that serves the `cache` directory. The server is setup as a Vite plugin in https://github.com/enso-org/engine-benchmark-results/blob/4e9f6bf8928a721a590347bb14f07b19694d84e7/vite.config.ts#L17 in dev mode.

In production, this server is not used, but the contents of the `cache` directory is copied inside the `dist` dir that is supposed to be served by the GH pages. Copy is implemented via `vite-plugin-static-copy` in https://github.com/enso-org/engine-benchmark-results/blob/4e9f6bf8928a721a590347bb14f07b19694d84e7/vite.config.ts#L21-L24

Commit:
Set MIN_DATE globally

Commit:
Update docs

JaroslavTulach commented on src/utils/data.ts:
```diff
@@ -0,0 +1,184 @@
+ * Fetches the index file.
+ */
+export async function loadIndex(): Promise<Index> {
+  const indexResp = await fetch(`${FS_URL}/cache/index.json`)
```
I don't see any `index.json` file in this PR!

That must mean this code is reading [already existing](https://github.com/enso-org/engine-benchmark-results/commit/7941e22441e3c597163e8dcae0936c4296356d43) `index.json` API in its already existing format.

Keeping the format compatible with currently existing one is good. Thanks.

Akirathan replied:
Yes, it is reading the already existing [cache/index.json](https://github.com/enso-org/engine-benchmark-results/blob/7941e22441e3c597163e8dcae0936c4296356d43/cache/index.json). There is nothing special about it - it is a mapping of file names to their timestamps.

Commit:
Merge branch 'main' into vue

# Conflicts:
#	engine-benchs.html
#	stdlib-benchs.html

JaroslavTulach reviewed (commented):
- Looks like `index.json` format remains unchanged. Good.
- keep existing `engine-benchs.html` and `stdlibs-benchs.html` files
- - keep links to them at main index page
- continue to regenerate the old HTML pages until we decide new UI is good enough

Commit:
Keep the old static websites deployed

Commit:
Update version in schema

Commit:
Links to old benchmarks are at the bottom of the page
